geode-cq/src/main/java/org/apache/geode/cache/query/cq/internal/command/CloseCQ.java

```diff
@@ -111,6 +112,13 @@ public void cmdExecute(final @NotNull Message clientMessage,
       sendCqResponse(MessageType.CQ_EXCEPTION_TYPE, "", clientMessage.getTransactionId(), cqe,
           serverConnection);
       return;
+    } catch (AuthenticationExpiredException e) {
+      if (serverConnection.getTransientFlag(REQUIRES_CHUNKED_RESPONSE)) {
```

Comment:
from line 61 of this command, this command does require chunked response, I think you can get rid of the if/else block and just send chunked exception. Same for the other command

Reply:
Okay, let me check and change ...

Reply:
After a look at it I'm wondering if it wouldn't be better to use the flag in deciding how to write the response. If at some point in the future the flag changes we'd also still have to change the code downstream if we don't read the flag when deciding how to write the exception.

Reply:
If they changed how the command should receive response, they would have to change how the command should send response. I don't think you need to complicate your code to do this pre-maturely.

Reply:
I understand that but why have the flag then?

Reply:
My guess is the flag is there to tell the other end that "I am sending my responses in chunked forms, so you need to prepare to receive it in such fashion".

Reply:
Okay